Local fs

jens1o opened the pull request:
fixes #21

jens1o commented:
@fiji-flo While writing the implementation tests I came up with the question what the expected behavior should be when we try to delete a file which can't be found. What do you think? I personally tend to ignore this kind of `std::io::ErrorKind::NotFound` error, but one could also argue that some implementation in the future may depend on knowing whether a file was previously there and deleted by the `Saver` implementation or wasn't there in the first place.

jens1o commented:
As I don't have access to S3 (nor real experience with it), I cannot really say what that implementation does in that case (it is not clearly documented) and the filesystem is not version based as S3 is.

fiji-flo commented on src/storage/saver/filesystem.rs:
```diff
@@ -0,0 +1,105 @@
+use std::sync::Arc;
+use uuid::Uuid;
+
+#[cfg(feature = "local-fs")]
```
We don't need this since it's only included in `mod.rs` if the feature is enabled.

fiji-flo commented on src/storage/saver/filesystem.rs:
```diff
@@ -0,0 +1,105 @@
+    pub path: Arc<PathBuf>,
+}
+
+#[cfg(feature = "local-fs")]
```
see above

fiji-flo commented on src/storage/saver/filesystem.rs:
```diff
@@ -0,0 +1,105 @@
+                Ok(()) => (),
+                // ignore error, as an error is also thrown when the directory already exists
+                // if the error is fatal, the write operation will also fail
+                Err(err) => info!(
```
log as `warn!`

fiji-flo reviewed (changes requested):
Looks great. Just a few minor things.

Also, please change the logging to start lowercase for consistency.

fiji-flo reviewed (approved):
:+1:

jens1o committed:
Introduce feature toggled local filesystem

The local fs is a drop-in replacement for the S3 based filesystem making
it easier for contributors to start hacking on it.

fixes #21

jens1o commented:
All (1) commits are now signed, please merge at your discretion. :)
